OpenNefia.Content/CharaMake/CharaMakeLayer.cs

```diff
@@ -0,0 +1,145 @@
+﻿using OpenNefia.Content.UI;
+using OpenNefia.Content.UI.Element;
+using OpenNefia.Content.UI.Element.Containers;
+using OpenNefia.Content.UI.Element.List;
+using OpenNefia.Core.Audio;
+using OpenNefia.Core.Locale;
+using OpenNefia.Core.Maths;
+using OpenNefia.Core.Rendering;
+using OpenNefia.Core.UI;
+using OpenNefia.Core.UI.Element;
+using OpenNefia.Core.UI.Layer;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenNefia.Content.CharaMake
+{
+    public class CharaMakeLayer : UiLayerWithResult<CharaMakeData, CharaMakeResult>, ICharaMakeLayer
+    {
+        protected class AttributeIcon : UiElement
+        {
+            private readonly Dictionary<string, string> _attributes = new Dictionary<string, string>
+            {
+                { "Elona.StatStrength", "0" },
+                { "Elona.StatConstitution", "1" },
+                { "Elona.StatDexterity", "2" },
+                { "Elona.StatPerception", "3" },
+                { "Elona.StatLearning", "4" },
+                { "Elona.StatWill", "5" },
+                { "Elona.StatMagic", "6" },
+                { "Elona.StatCharisma", "7" }
+
+            };
+            private IAssetInstance AssetAttributeIcons;
+            private string Type;
+
+            public AttributeIcon(string type)
+            {
+                AssetAttributeIcons = Assets.Get(AssetPrototypeOf.AttributeIcons);
+                Type = type;
+            }
+
+            public override void Draw()
+            {
+                base.Draw();
+                GraphicsEx.SetColor(Color.White);
+                if (_attributes.TryGetValue(Type, out var iconId))
+                    AssetAttributeIcons.DrawRegion($"{iconId ?? "2"}", X, Y, centered: true);
+            }
+
+            public override void GetPreferredSize(out Vector2i size)
+            {
+                size = new Vector2i(10, 10);
+            }
+        }
+
+        protected IAssetInstance AssetBG;
+        private IAssetInstance[] AssetWindows;
+
+        protected CharaMakeData Data = default!;
+        private UiTopicWindow CaptionWindow;
+        [Localize]
+        protected UiText Caption;
+
+        protected readonly string[] AttributeIds = new[]
+        {
+            "Elona.StatStrength",
+            "Elona.StatConstitution",
+            "Elona.StatDexterity",
+            "Elona.StatPerception",
+            "Elona.StatLearning",
+            "Elona.StatWill",
+            "Elona.StatMagic",
+            "Elona.StatCharisma"
+        };
+
+        public CharaMakeLayer()
+        {
+            AssetBG = Assets.Get(AssetPrototypeOf.Void);
+
+            AssetWindows = new[]
```

Comment:
I don't think these are getting drawn. They should appear below the list of options, like this.

![2022-01-06 18_16_57-Elona+ Custom-GX 2 08 1 0](https://user-images.githubusercontent.com/6700637/148480643-92cfbb57-8258-4efc-91f3-184bd5f33fc3.png)

Reply:
yeah I wanted to get the functionality done first, I'll see if I can get them added.

Reply:
done in dd52918